packages/components/src/custom-gradient-picker/utils.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * External dependencies
+ */
+import gradientParser from 'gradient-parser';
+import { colord, extend } from 'colord';
+import namesPlugin from 'colord/plugins/names';
+
+/**
+ * Internal dependencies
+ */
+import {
+	DEFAULT_GRADIENT,
+	HORIZONTAL_GRADIENT_ORIENTATION,
+	DIRECTIONAL_ORIENTATION_ANGLE_MAP,
+} from './constants';
+import { serializeGradient } from './serializer';
+import type { ControlPoint } from './types';
+
+extend( [ namesPlugin ] );
+
+export function getLinearGradientRepresentation(
+	gradientAST: gradientParser.GradientNode
+) {
+	return serializeGradient( {
+		type: 'linear-gradient',
+		orientation: {
+			...HORIZONTAL_GRADIENT_ORIENTATION,
+			value: `${ HORIZONTAL_GRADIENT_ORIENTATION.value }`,
+		},
+		colorStops: gradientAST.colorStops,
+		// NTS: As noted elsewhere, gradientParser's DefinetelyTyped values provide no overlap between the different `orientation` values for linear vs radial nodes.
+	} as gradientParser.LinearGradientNode );
```

Comment:
As noted elsewhere, gradientParser's `DefinetelyTyped` values provide no overlap between the different `orientation` values for linear vs radial nodes.